### ISSUE TITLE
Error generating JS with nodejs_grpc_compile

### ISSUE THREAD

**Issue by ewang** [rules-proto-grpc/rules_proto_grpc#60]
I have the following BUILD definition:

```
proto_library(
    name = "example_proto",
    srcs = [
        "example_api.proto",
    ],
    deps = [
        "@com_google_googleapis//google/api:annotations_proto",
    ],
)

nodejs_grpc_compile(
    name = "example_js_grpc",
    deps = [":example_proto"],
)
```

This is the error I'm getting:

```
ERROR: /private/var/tmp/_bazel/fe3b81392cdf65e8a52f85ec1d386a65/external/com_google_googleapis/google/api/BUILD.bazel:9:1: output 'external/com_google_googleapis/google/api/annotations_proto/nodejs_grpc_compile_aspect_verb0/google/api/annotations_grpc_pb.js' was not created
ERROR: /private/var/tmp/_bazel/fe3b81392cdf65e8a52f85ec1d386a65/external/com_google_googleapis/google/api/BUILD.bazel:9:1: not all outputs were created or valid
ERROR: /private/var/tmp/_bazel/fe3b81392cdf65e8a52f85ec1d386a65/external/com_google_protobuf/BUILD:284:2: output 'external/com_google_protobuf/descriptor_proto/nodejs_grpc_compile_aspect_verb0/google/protobuf/descriptor_grpc_pb.js' was not created
ERROR: /private/var/tmp/_bazel/fe3b81392cdf65e8a52f85ec1d386a65/external/com_google_protobuf/BUILD:284:2: not all outputs were created or valid
ERROR: /private/var/tmp/_bazel/fe3b81392cdf65e8a52f85ec1d386a65/external/com_google_googleapis/google/api/BUILD.bazel:101:1: output 'external/com_google_googleapis/google/api/http_proto/nodejs_grpc_compile_aspect_verb0/google/api/http_grpc_pb.js' was not created
ERROR: /private/var/tmp/_bazel/fe3b81392cdf65e8a52f85ec1d386a65/external/com_google_googleapis/google/api/BUILD.bazel:101:1: not all outputs were created or valid
```

Those files in question don't have service definitions.

**Comment by alisonbelow** [comment]
I am also seeing this issue, but only with the compilation of well-known protos (e.g. descriptor like seen above). If I turn on `debug = 4` to the `nodejs_grpc_compile` rule, I can see:
```
##### SANDBOX BEFORE RUNNING PROTOC
./bazel-out/host/bin/external/com_google_protobuf/protoc
./bazel-out/darwin-opt/bin/external/com_google_protobuf/descriptor_proto-descriptor-set.proto.bin

##### SANDBOX AFTER RUNNING PROTOC
./bazel-out/darwin-opt/bin/external/com_google_protobuf/descriptor_proto/nodejs_grpc_compile_aspect_verb4/google/protobuf/descriptor_pb.js

##### SANDBOX BEFORE RUNNING PROTOC
./bazel-out/host/bin/external/com_google_protobuf/protoc
./bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_github_grpc_grpc/src/compiler/grpc_node_plugin
./bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_github_grpc_grpc/src/compiler/grpc_node_plugin.runfiles/REPO/external/com_github_grpc_grpc/src/compiler/grpc_node_plugin
./bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/com_github_grpc_grpc/src/compiler/grpc_node_plugin.runfiles/com_github_grpc_grpc/src/compiler/grpc_node_plugin
./bazel-out/darwin-opt/bin/external/com_google_protobuf/descriptor_proto-descriptor-set.proto.bin

##### SANDBOX AFTER RUNNING PROTOC
ERROR: /private/var/tmp/HASH/external/com_google_protobuf/BUILD:319:2: output 'external/com_google_protobuf/descriptor_proto/nodejs_grpc_compile_aspect_verb4/google/protobuf/descriptor_grpc_pb.js' was not created
ERROR: /private/var/tmp/HASH/external/com_google_protobuf/BUILD:319:2: not all outputs were created or valid
```

It seems like expectation of compiling well known protos with grpc_node_plugin tool should be removed, as that is not a reasonable expected output

**Comment by aaliddell** [comment]
Could you try the version in the 'well-known-node' branch? This adds the well-known protos to the exclusions for the plugins.

https://github.com/rules-proto-grpc/rules_proto_grpc/tree/well-known-node

**Comment by alisonbelow** [comment]
@aaliddell Tried patching release 1.0.2 with attached (added .txt extension to upload). Did this because the branch you linked but too far behind master, needed a repos macro load in //:repostories.bzl.

[rules_proto_grpc-nodejs_BUILD-1.0.2.patch.txt](https://github.com/rules-proto-grpc/rules_proto_grpc/files/4389117/rules_proto_grpc-nodejs_BUILD-1.0.2.patch.txt)

I get errors because not every protobuf file I am compiling contains a grpc service definition, which is half of this issue, potentially. I no longer get errors for the well known protos. Instead of just creating empty <proto>_grpc_pb.js, I get errors for each *.proto:

```
ERROR: /path/to/BUILD:LINE:CHAR: output 'path/to/nodejs_grpc_compile_aspect_verb4/<proto>_grpc_pb.js' was not created
```

Will consider this a fix for half of @ewang's issue. Maybe this is a feature, not a bug, but the `nodejs_grpc_compile` rule returns an error for each one of the proto files that does not contains a grpc Service definition.

**Comment by Smokrow** [comment]
I am finding myself having the same issue. Any news on how to fix this?

**Comment by Smokrow** [comment]
It looks like the Rule expects that a *_grpc_pb.js to be created, which of course does not work when there is no service definition in this file. This is happening as soon as you import another protobuf file without service definitions. 

Any idea where to look at? I could help out on this :)

**Comment by alisonbelow** [comment]
Was able to get my build functional with this patch (a combination of above and another fix):
```
--- nodejs/BUILD.bazel.old	2020-04-14 15:35:47.000000000 -0700
+++ nodejs/BUILD.bazel	2020-04-14 15:39:28.000000000 -0700
@@ -7,13 +7,23 @@
         "import_style=commonjs",
         "binary",
     ],
     outputs = ["{protopath}_pb.js"],
+    exclusions = [
+        "google/protobuf",
+    ],
     visibility = ["//visibility:public"],
 )
 
 proto_plugin(
     name = "grpc_nodejs_plugin",
-    outputs = ["{protopath}_grpc_pb.js"],
+    # Do not expect *_grpc_pb.js output for every file
+    # This assumes that every input protobuf contains Service definition
+    #outputs = ["{protopath}_grpc_pb.js"],
+    # Use output_directory arg to remove expectations on output files each time target is built
+    output_directory = True,
     tool = "@com_github_grpc_grpc//src/compiler:grpc_node_plugin",
+    exclusions = [
+        "google/protobuf",
+    ],
     visibility = ["//visibility:public"],
 )
```

**Comment by bnczk** [comment]
This issue is similar to #58

**Comment by aaliddell** [comment]
I thought there was a reason `output_directory` wasn't suitable with rules_nodejs, but I'll try that patch and see how the tests look.

**Comment by aaliddell** [comment]
Fixed in #84
